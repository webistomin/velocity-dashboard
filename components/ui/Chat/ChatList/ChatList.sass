+b('chat-list')
  background-color: rgb(var(--color-white))

  +e('item')
    border-bottom: rem(1px) solid rgba(var(--color-primary), 0.2)

  +e('link')
    display: flex
    box-sizing: border-box
    width: 100%
    height: 100%
    padding: rem(20px) rem(14px)
    transition: background-color 0.2s ease-out
    position: relative

    &::before
      content: ''
      width: 100%
      height: 100%
      position: absolute
      top: 0
      left: 0
      z-index: 0
      background-color: rgb(var(--color-primary))
      opacity: 0
      transition: opacity 0.2s ease-out, background-color 0.2s ease-out

    &:hover::before
      opacity: 0.15

    +media($screen-sm)
      padding-left: rem(24px)
      padding-right: rem(24px)

  +e('col')
    +m('left')
      position: relative
      margin-right: rem(9px)

    +m('right')
      width: 100%
      display: flex
      flex-direction: column
      min-width: 0

  +e('heading')
    display: flex
    justify-content: space-between
    align-items: baseline
    margin-bottom: rem(4px)

  +e('message')
    +text-overflow()
    max-width: 100%

  +e('name')
    font: var(--text-button)

  +e('time')
    font: var(--font-caption)
    color: rgb(var(--color-gray-2))
    +text-overflow()
    max-width: 100%

  +e('message')
    font: var(--font-caption)
    color: rgb(var(--color-gray-2))
