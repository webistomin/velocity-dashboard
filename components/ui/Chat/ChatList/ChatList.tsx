import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';
import { IChatFormatted } from 'common/types/chat/chat-list';
import { VNode } from 'vue';

import BaseThumbnail from 'components/base/BaseThumbnail';
import BaseNotify from 'components/base/BaseNotify';

import './ChatList.sass';

export interface IChatListProps {
  chats: IChatFormatted[] | null;
}

@Component({
  name: 'ChatList',
})
export default class ChatList extends VueComponent<IChatListProps> {
  @Prop({ default: null })
  private readonly chats!: IChatListProps['chats'];

  public render(): VNode {
    const chatsList = this.chats
      ? this.chats.map((chat) => {
          return (
            <li class='chat-list__item list-item'>
              <nuxt-link class='chat-list__link link' to={`/chat/${chat.id}`}>
                <span class='chat-list__col chat-list__col_left'>
                  <BaseNotify />
                  <BaseThumbnail image={chat.avatar} alt={chat.name} size='m' />
                </span>
                <span class='chat-list__col chat-list__col_right'>
                  <span class='chat-list__heading'>
                    <strong class='chat-list__name'>{chat.name}</strong>
                    <time class='chat-list__time'>{chat.formattedTime}</time>
                  </span>
                  <span class='chat-list__message'>{chat.message}</span>
                </span>
              </nuxt-link>
            </li>
          );
        })
      : null;
    return (
      <div class='chat-list'>
        <ul class='chat-list__list list'>{chatsList}</ul>
      </div>
    );
  }
}
