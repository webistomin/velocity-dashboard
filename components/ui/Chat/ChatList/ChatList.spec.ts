import { shallowMount } from '@vue/test-utils';
import ChatList from './ChatList';

describe('ChatList', () => {
  const ItemSelector = '.chat-list__item';

  const factory = (propsData = {}, options = {}) => {
    return shallowMount(ChatList, {
      propsData: {
        chats: [
          {
            name: 'Alexey Istomin',
            avatar: '/img/avatar.png',
            message: 'You are most welcome Kathey! You are most welcome Kathey! You are most welcome Kathey!',
            id: 'abc132',
            time: '02/06/2020',
          },
        ],
        ...propsData,
      },
      stubs: ['nuxt-link'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Have no children if chats is not provided', () => {
    const wrapper = factory({
      chats: null,
    });
    const item = wrapper.find(ItemSelector);
    expect(item.exists()).toEqual(false);
  });
});
