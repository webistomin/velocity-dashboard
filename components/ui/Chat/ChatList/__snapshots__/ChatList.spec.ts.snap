// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList Matches snapshot 1`] = `
"<div class=\\"chat-list\\">
  <ul class=\\"chat-list__list list\\">
    <li class=\\"chat-list__item list-item\\">
      <nuxt-link-stub to=\\"/chat/abc132\\" class=\\"chat-list__link link\\"><span class=\\"chat-list__col chat-list__col_left\\"><vuecomponent-stub></vuecomponent-stub><vuecomponent-stub image=\\"/img/avatar.png\\" alt=\\"Alexey Istomin\\" size=\\"m\\"></vuecomponent-stub></span><span class=\\"chat-list__col chat-list__col_right\\"><span class=\\"chat-list__heading\\"><strong class=\\"chat-list__name\\">Alexey Istomin</strong><time class=\\"chat-list__time\\"></time></span><span class=\\"chat-list__message\\">You are most welcome Kathey! You are most welcome Kathey! You are most welcome Kathey!</span></span></nuxt-link-stub>
    </li>
  </ul>
</div>"
`;
